backend/src/main/java/se/kth/iv1201/recruitmentapplicationgroup5/controller/ApiExceptionHandler.java

```diff
@@ -0,0 +1,75 @@
+package se.kth.iv1201.recruitmentapplicationgroup5.controller;
+
+import javax.validation.ConstraintViolationException;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+
+/**
+ * Class for encapsulating non default exception handling in one place.
+ *
+ */
+@ControllerAdvice
+public class ApiExceptionHandler {
+	
+	//Logger here as a private field
+	
+	@ExceptionHandler(value = MethodArgumentNotValidException.class)
+	public ResponseEntity<Object> handleValidationException(
+			MethodArgumentNotValidException e, 
+			WebRequest req)
+	{
+		var status = HttpStatus.BAD_REQUEST;
```

Comment:
Using **var** is cool and I like it, but keep in mind that it is nothing that we have discussed. Is that something we should always do when possible?

Reply:
I'm not using it like that. I could've used it for all variables here. I used var when it was clear which type the variable would have, or when you don't want to expose some strange custom type.